runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -221,6 +226,86 @@ private ClusterStatus getClusterHealth(Layout layout, Set<String> peerResponsive
+            } catch (Exception e) {
+                log.error("getClusterStatus: Exception encountered connecting to {}. ",
+                        server, e);
+                CompletableFuture<Layout> cf = new CompletableFuture<>();
+                cf.completeExceptionally(e);
+                layoutFuturesMap.put(server, cf);
+            }
+        }
+
+        // Wait on the Completable futures and retain the layout with the highest epoch.
+        for (Entry<String, CompletableFuture<Layout>> entry : layoutFuturesMap.entrySet()) {
+            try {
+                Layout nodeLayout = entry.getValue().get();
+
+                if (layout == null || nodeLayout.getEpoch() > layout.getEpoch()) {
+                    layout = nodeLayout;
+                }
+            } catch (InterruptedException ie) {
+                throw new UnrecoverableCorfuInterruptedError(ie);
+            } catch (ExecutionException ee) {
+                log.error("getClusterStatus: Error while fetching layout from {}.",
+                        entry.getKey(), ee);
+            }
+        }
+        return layout;
+    }
+
+    /**
+     * Create node status map.
+     * Assigns a node status to all the nodes in the layout.
+     * UP: If the node is responsive and present in all the segments OR present in none (for cases
+     * where a node is a Layout or Sequencer Server only and does not possess a LogUnit component.)
+     * DB_SYNCING: If the node is present in a few segments but not all. This node is syncing data.
+     * DOWN: If the node is unresponsive.
+     *
+     * @param layout            Layout used to compute cluster status
+     * @param responsiveServers Set of responsive servers excluding unresponsive Servers list.
+     * @param allServers        All servers in the layout.
+     * @return Map of nodes mapped to their status.
+     */
+    private Map<String, NodeStatus> createNodeStatusMap(Layout layout,
+                                                        Set<String> responsiveServers,
+                                                        Set<String> allServers) {
+        Map<String, NodeStatus> nodeStatusMap = new HashMap<>();
+        final int segmentsCount = layout.getSegments().size();
+        for (String server : allServers) {
+            if (responsiveServers.contains(server)
+                    && !layout.getUnresponsiveServers().contains(server)) {
+                int nodeInSegments = 0;
+                for (LayoutSegment layoutSegment : layout.getSegments()) {
+                    if (layoutSegment.getAllLogServers().contains(server)) {
+                        nodeInSegments++;
+                    }
+                }
+                // If the node is not present in an
+                nodeStatusMap.put(server, nodeInSegments == 0 || nodeInSegments == segmentsCount
+                        ? NodeStatus.UP : NodeStatus.DB_SYNCING);
+            }
+            nodeStatusMap.putIfAbsent(server, NodeStatus.DOWN);
```

Comment:
nit: might be more readable to have a variable for nodeStatus (with default of DOWN) which will be updated in the if clause and then have one statement of put on nodeStatusMap.